gtk4/src/subclass/editable.rs

```diff
@@ -4,10 +4,68 @@ use super::widget::WidgetImpl;
 use crate::Editable;
 use glib::subclass::prelude::*;
 use glib::translate::*;
-use glib::{Cast, GString};
+use glib::{Cast, GString, ObjectExt, Quark};
 use libc::{c_char, c_int};
+use once_cell::sync::Lazy;
 
 pub trait EditableImpl: WidgetImpl {
+    fn properties(&self) -> &'static [String] {
```

Comment:
This is definitely not very nice, but see the example on how I handled it :) it's probably the only way to make it work without too much involvement from the user.

Reply:
Well, you could also make it a `&'static [&'static str]` as they're all string literals, or go one step further and make it a `&[glib::ParamSpec]` :)

Reply:
You could also add a function that checks if the pspec from set/get_property() is one that is handled by this interface. That might be even easier for the user

Reply:
Also all these 3 things should be in an `EditableImplExt` trait. Nobody should ever override that

Reply:
Also you should dynamically get these properties instead of hardcoding, or it will fall apart if a new property is added at some point

Reply:
Actually, I did something much nicer :-) I stored the range of the props ids that was added at `install_properties` stage and provided a function to check whether the id is in the range of the added props or not.